File: romancalculator/javascript/spec/romanToNumberSpec.js

describe("romanToNumber", function () {
  it("vide donne 0", function () {
    expect(romanToNumber("")).toEqual(0);
  });

  it("I donne 1", function () {
    expect(romanToNumber("I")).toEqual(1);
  });

  it("V donne 5", function () {
    expect(romanToNumber("V")).toEqual(5);
  });

  it("VI donne 6", function () {
    expect(romanToNumber("VI")).toEqual(6);
  });
});

File: spec/roman_to_number_spec.rb
require_relative '../roman_to_number'

RSpec.describe "Une fonction de conversion de nombre romain en nombre arabe" do
  it "transforme I en 1 " do
    valeur = 'I'
    valeur_attendue = 1
    resultat = transforme(valeur)
    expect(resultat).to eq(valeur_attendue)
  end

  it "transforme V en 5 " do
    valeur = 'V'
    valeur_attendue = 5
    resultat = transforme(valeur)
    expect(resultat).to eq(valeur_attendue)
  end
end